### PR TITLE
Use shared resources in charge-deployment to T

### PR DESCRIPTION
**Pull request by bemwen**
## Description

The purpose of this PR is to use shared resources when receiving metering point created events.

Besides this, some housekeeping has been done to align our setup more with the strategy used in shared resources.

This PR includes:

* Removing stub in T-environment and use shared resources instead
* Changes how our stub in development environments work, to align it more with the strategy used in the shared resources repository
* Moves fixed strings to the locals.tf file instead of having these in separate tf-files which might get out of sync
* Uses the general listen and sender strings made available in shared resources instead of specific ones for metering point created topic
* Splits some stub files to have a 1-to-1 reflection of shared resources, hopefully making it easier to compare later
* Upgrades all modules used in our pipeline to 1.7.0

## References

* #359

### CODE REVIEW

File: build/infrastructure/env/Development/kv-shared-stub.tf [hunk 1]
```diff
@@ -53,3 +53,19 @@ module "kv_shared_stub" {
     }
   ]
 }
+
+module "kvs_integrationevents_listener_connection_string" {
+  source        = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//key-vault-secret?ref=1.6.0"
```
Comment (HenrikSommer) [inline build/infrastructure/env/Development/kv-shared-stub.tf:58]: Use version 1.7.0

File: build/infrastructure/env/Development/kv-shared-stub.tf [hunk 2]
```diff
@@ -53,3 +53,19 @@ module "kv_shared_stub" {
...
+  name          = local.INTEGRATION_EVENTS_LISTENER_CONNECTION_STRING
+  value         = module.sbnar_integrationevents_listener.primary_connection_string
+  key_vault_id  = module.kv_shared_stub.id
+  dependencies  = [module.sbnar_integrationevents_listener.dependent_on]
+}
+
+module "kvs_integrationevents_sender_connection_string" {
+  source        = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//key-vault-secret?ref=1.6.0"
```
Comment (HenrikSommer) [inline build/infrastructure/env/Development/kv-shared-stub.tf:66]: Use version 1.7.0

**Review by lasrinnil — APPROVED** [review]

File: build/infrastructure/env/Development/kv-shared-stub.tf (hunk 1 above)
Comment (bemwen) [reply build/infrastructure/env/Development/kv-shared-stub.tf]: Nice catch, changed

File: build/infrastructure/env/Development/kv-shared-stub.tf (hunk 2 above)
Comment (bemwen) [reply build/infrastructure/env/Development/kv-shared-stub.tf]: Changed

File: build/infrastructure/main/azfun-metering-point-created-receiver.tf [hunk 3]
```diff
@@ -30,7 +30,7 @@ module "azfun_metering_point_created_receiver" {
     WEBSITES_ENABLE_APP_SERVICE_STORAGE                = true
     FUNCTIONS_WORKER_RUNTIME                           = "dotnet"
 
-    METERING_POINT_CREATED_LISTENER_CONNECTION_STRING  = data.azurerm_key_vault_secret.metering_point_created_listener_connection_string.value
+    METERING_POINT_CREATED_LISTENER_CONNECTION_STRING  = data.azurerm_key_vault_secret.integration_events_listener_connection_string.value
```
Comment (HenrikSommer) [inline build/infrastructure/main/azfun-metering-point-created-receiver.tf:33]: Maybe rename the variable to INTEGRATION_EVENTS_LISTENER_CONNECTION_STRING?

Comment (bemwen) [reply build/infrastructure/main/azfun-metering-point-created-receiver.tf:33]: That would break the function, so that should be changed in a separate PR

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**